arkouda/pdarraycreation.py

```diff
@@ -147,38 +172,39 @@ def array(a: Union[pdarray, np.ndarray, Iterable], dtype: Union[np.dtype, type,
     otherwise a RuntimeError will be raised. This is to protect the user
     from overwhelming the connection between the Python client and the arkouda
     server, under the assumption that it is a low-bandwidth connection. The user
-    may override this limit by setting ak.maxTransferBytes to a larger value, 
+    may override this limit by setting ak.maxTransferBytes to a larger value,
     but should proceed with caution.
-    
-    If the pdrray or ndarray is of type U, this method is called twice recursively 
-    to create the Strings object and the two corresponding pdarrays for string 
+
+    If the pdrray or ndarray is of type U, this method is called twice recursively
+    to create the Strings object and the two corresponding pdarrays for string
     bytes and offsets, respectively.
 
     Examples
     --------
     >>> ak.array(np.arange(1,10))
     array([1, 2, 3, 4, 5, 6, 7, 8, 9])
-    
+
     >>> ak.array(range(1,10))
     array([1, 2, 3, 4, 5, 6, 7, 8, 9])
-   
-    >>> strings = ak.array(['string {}'.format(i) for i in range(0,5)])
+
+    >>> strings = ak.array([f'string {i}' for i in range(0,5)])
     >>> type(strings)
-    <class 'arkouda.strings.Strings'>  
+    <class 'arkouda.strings.Strings'>
     """
     # Removed @typechecked to prevent cyclic dependencies with ArrayView
     from arkouda.numeric import cast as akcast
+
     # If a is already a pdarray, do nothing
     if isinstance(a, pdarray):
         return a if dtype is None else akcast(a, dtype)
     from arkouda.client import maxTransferBytes
+
     # If a is not already a numpy.ndarray, convert it
     if not isinstance(a, np.ndarray):
         try:
             a = np.array(a, dtype=dtype)
-        except:
-            raise TypeError(('a must be a pdarray, np.ndarray, or convertible to' +
-                            ' a numpy array'))
+        except (RuntimeError, TypeError, ValueError):
+            raise TypeError("a must be a pdarray, np.ndarray, or convertible to a numpy array")
```

Comment:
For the bare excepts, I agree `except Exception` is the way to go. For the multiple types, if we know that those are the types we are expecting and do not want to handle any others specifically, I am not sure we should switch to the more general `except Exception` because we may end up catching exception cases we don't want. However, if we have them catching specific ones because those are what we *think* will be raised, we should switch to the general case.

Reply:
I agree, if it's there specifically because we want these types to be handled and others to fail normally then we should keep it. This was the only other spot I noticed that was using this format, so I thought I would bring it up for consideration in case it's meant to just raise that `TypeError` with any Exception